backend/src/main/java/eu/bbmri_eric/negotiator/negotiation/NegotiationController.java

```diff
@@ -306,12 +321,72 @@ public CollectionModel<EntityModel<ResourceWithStatusDTO>> updateResources(
   @SecurityRequirement(name = "security_auth")
   public ResponseEntity<byte[]> generateNegotiationPdf(
-      @Valid @PathVariable String id,
-      @RequestParam(value = "template", required = false) String templateName) {
+      @PathVariable String id,
+      @Parameter(
+              description =
+                  "Specific template to be used for generation, identified by the template name. If omitted the default template is used.")
+          @RequestParam(value = "template", required = false)
+          String templateName)
+      throws Exception {
+
+    byte[] pdfBytes = generateNegotiationPdfInternal(id, templateName);
+    return ResponseEntity.ok()
+        .contentType(MediaType.APPLICATION_PDF)
+        .header("Content-Disposition", "attachment; filename=\"negotiation-" + id + ".pdf\"")
+        .body(pdfBytes);
+  }
+
+  @GetMapping(value = "/negotiations/{id}/fullpdf", produces = MediaType.APPLICATION_PDF_VALUE)
+  @Operation(
+      summary =
+          "Generate a PDF for a negotiation including all attachments. converted to PDF format, if the source document format is supported for conversion.",
+      description =
+          "This endpoint creates a merged PDF including all attached documents if the source document format is supported for conversion to PDF. A specific template can be applied to generate the PDF.")
+  @SecurityRequirement(name = "security_auth")
+  public ResponseEntity<byte[]> generateNegotiationPdfWithAttachments(
+      @PathVariable String id,
+      @Parameter(
+              description =
+                  "Specific template to be used for generation, identified by the template name. If omitted the default template is used.")
+          @RequestParam(value = "template", required = false)
+          String templateName)
+      throws Exception {
+
+    byte[] negotiationPdf = generateNegotiationPdfInternal(id, templateName);
+    List<String> attachmentIds =
+        attachmentService.findByNegotiation(id).stream().map(AttachmentMetadataDTO::getId).toList();
+
+    List<byte[]> pdfsToMerge = new java.util.ArrayList<>();
+    pdfsToMerge.add(negotiationPdf);
+
+    if (!attachmentIds.isEmpty()) {
+      List<byte[]> attachmentPdfs = mergingService.getAttachmentsAsPdf(attachmentIds);
+      pdfsToMerge.addAll(attachmentPdfs);
+    }
+
+    byte[] mergedPdf = null;
+    try {
+      mergedPdf = PdfMerger.mergePdfs(pdfsToMerge);
+    } catch (IOException e) {
+      throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, "Error merging PDFs", e);
+    }
+
+    HttpHeaders headers = new HttpHeaders();
+    headers.setContentType(MediaType.APPLICATION_PDF);
+    headers.setContentDisposition(
+        ContentDisposition.builder("attachment")
+            .filename(String.format("%s_merged.pdf", id))
+            .build());
+    return new ResponseEntity<>(mergedPdf, headers, HttpStatus.OK);
+  }
```

Comment:
would suggest moving all this logic to a service level. The controller should IMHO just validate input, call the service and then produce output